src/components/ReportActionItem/TaskPreview.js

```diff
@@ -74,13 +77,7 @@ function TaskPreview(props) {
                     containerStyle={[styles.taskCheckbox]}
                     isChecked={isTaskCompleted}
                     disabled={TaskUtils.isTaskCanceled(props.taskReport)}
-                    onPress={() => {
-                        if (isTaskCompleted) {
-                            TaskUtils.reopenTask(props.taskReportID, taskTitle);
-                        } else {
-                            TaskUtils.completeTask(props.taskReportID, taskTitle);
-                        }
-                    }}
+                    onPress={() => (isTaskCompleted ? TaskUtils.reopenTask(props.taskReport) : TaskUtils.completeTask(props.taskReport))}
```

Comment:
this might be a good spot to use the hook `useCallback` https://react.dev/reference/react/useCallback

`isTaskCompleted` and `taskReport` would be the dependencies, i think

Reply:
Cool, I'll check that in a bit! Thks for the suggestion

Reply:
I think it will be a waste in this case. The Checkbox will always re-render since we don't use `React.memo`.

If we need to optimise this we will have first to optimise Checkbox to use `React.memo` using a custom [`arePropsEqual`](https://react.dev/reference/react/memo) callback, e.g.:
```js
export default React.memo(Checkbox, (prevProps, nextProps) => {
    if (prevProps.isChecked !== nextProps.isChecked) {
        return false;
    }
    if deep compare style not equal return false;
    if deep compare containerStyle not equal return false;
    etc.
    return true;
});
```